tests/compliance/blackduck-spec.js

```diff
@@ -0,0 +1,176 @@
+            { text: 'blargh' }
+          ]
+        }
+      }
+      const expected = 'MIT, Apache, UNKNOWN'
+      expect(blackduck._getLicense(licenseSources)).to.eql(expected)
+    })
+  })
+
+  describe('gets npm license data', function () {
+    let findLicenseStub
+    let writeFileStub
+
+    beforeEach(function () {
+      findLicenseStub = sandbox.stub(blackduck, 'findLicenses')
+      findLicenseStub.returns(Promise.resolve([
+        {
+          id: 'some-id',
+          name: 'some-package-1',
+          version: 1.0,
+          licenseSources: undefined
+        },
+        {
+          id: 'some-other-id',
+          name: 'some-package-2',
+          version: 2.0,
+          licenseSources: {
+            package: {
+              sources: []
+            },
+            license: {
+              sources: [
+                { text: 'MIT' },
+                { text: 'Apache' },
+                { text: 'blargh' }
+              ]
+            }
+          }
+        }
+      ]))
+      writeFileStub = sandbox.stub(blackduck, 'writeFile')
+      writeFileStub.returns(Promise.resolve())
+    })
+
+    it('and logs if there is none', function () {
+      return blackduck.getNpmLicenseData('/some/path', '/some/output/path').then(() => {
+        expect(logger.log.firstCall.args).to.eql(['ERROR: some-id has no licenseSources?'])
+      })
+    })
+
+    it('attempts to write file with correct data', function () {
+      const expectedData = [
+        {
+          version: 2.0,
+          name: 'some-package-2',
+          license: 'MIT, Apache, UNKNOWN'
+        }
+      ]
+      return blackduck.getNpmLicenseData('/some/path', '/some/output/path').then((value) => {
+        expect(writeFileStub.lastCall.args).to.eql([
+          '/some/output/path',
+          JSON.stringify({
+            'schema-version': 1,
+            data: expectedData
+          }, null, '  ')
+        ])
+      })
+    })
+
+    it('returns file written message', function () {
+      return blackduck.getNpmLicenseData('/some/path', '/some/output/path').then((value) => {
+        expect(value).to.eql('successfully wrote /some/output/path')
+      })
+    })
+
+    it('logs file write error', function () {
+      writeFileStub.returns(Promise.reject('some error'))
+      return blackduck.getNpmLicenseData('/some/path', '/some/output/path').catch(() => {
+        expect(logger.log.lastCall.args).to.eql(['(1) ERROR: writing /some/output/path', 'some error'])
+      })
+    })
+  })
+
+  describe('gets package data', function () {
+    let readFileStub
+    const fileData = ' "ember-frost-1"\n "frost-1"'
+    beforeEach(function () {
+      readFileStub = sandbox.stub(blackduck, 'readFile')
+      readFileStub.returns(Promise.resolve(fileData))
+    })
+    it('returns the correct repo paths', function () {
+      const expected = 'https://bitbucket.ciena.com/scm/bp_frost/frost-1.git\nhttps://github.com/ciena-frost/ember-frost-1.git\n'
```

Comment:
Once the regexes and URLs for those regexes are updated to be in `.pr-bumper.json`, we'll want a test that checks for overriding them as well.